cmd/commands/datasetIngestor.go

```diff
@@ -68,14 +68,57 @@ For Windows you need instead to specify -user username:password on the command l
 		userpass, _ := cmd.Flags().GetString("user")
 		token, _ := cmd.Flags().GetString("token")
 		copyFlag, _ := cmd.Flags().GetBool("copy")
-		nocopyFlag, _ := cmd.Flags().GetBool("nocopy") // TODO why is there even a "no copy" flag?
+		nocopyFlag, _ := cmd.Flags().GetBool("nocopy")
+		transferTypeFlag, _ := cmd.Flags().GetString("transfer-type")
 		tapecopies, _ := cmd.Flags().GetInt("tapecopies")
 		autoarchiveFlag, _ := cmd.Flags().GetBool("autoarchive")
 		linkfiles, _ := cmd.Flags().GetString("linkfiles")
 		allowExistingSourceFolder, _ := cmd.Flags().GetBool("allowexistingsource")
 		addAttachment, _ := cmd.Flags().GetString("addattachment")
 		addCaption, _ := cmd.Flags().GetString("addcaption")
 		showVersion, _ := cmd.Flags().GetBool("version")
+		globusCfgFlag, _ := cmd.Flags().GetString("globus-cfg")
+
+		// TODO: read in CFG!
+
+		// transfer type
+		transferType, err := convertToTransferType(transferTypeFlag)
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+		var transferFiles func(params cliutils.TransferParams) (archivable bool, err error)
+
+		// globus specific vars (if needed)
+		var globusClient globus.GlobusClient
+		var srcCollection, srcPrefixPath, destCollection, destPrefixPath string
+
+		switch transferType {
```

Comment:
Shouldn't this configuration be skipped if copying is not required?

Reply:
I could add an if block around this. My thinking was that I handle the transfer selection in advance so that the rest of the code can operate in the same way with minimal changes.

Reply:
This might have been a bad suggestion (see below)